### DISCUSSION TITLE
New installation and restore backup not work

### DISCUSSION THREAD

**Discussion by Pindol83** [firefly-iii/firefly-iii#8101] [Firefly III]
Good morning folks,

I want to mention that I'm not very experienced with Docker, and I'm currently learning. I've installed Firefly as a container on MacOS and have performed the backup as described in the documentation: https://docs.firefly-iii.org/firefly-iii/advanced-installation/backup/

Today, I tried to create a new installation on a Debian virtual machine and restore the backup of the other database (I copied and pasted the same files I use on MacOS: docker-compose.yml, .env, and .db.env and i've completed the installation with these files.).

I created the user with the same name and password as the one I have on Firefly in MacOS then i restored the backup file as per the procedure using the following command (adjusted for my folders):

```
docker run --rm -v "firefly_iii_db:/recover" \
    -v "$HOME/backups/firefly:/backup" \
    ubuntu tar -xvf /backup/firefly_db.tar -C /recover --strip 1
```
But it doesn't seem to be happening, after running this command, how long should I wait?

Where am I going wrong?
Thank you.

**Comment by JC5** [comment]
I'm not sure what you mean, it should just take a few minutes though.

**Comment by Pindol83** [comment]
> I'm not sure what you mean, it should just take a few minutes though.

I waited for more than 10 minutes, but nothing worked with manual backup restoration.

In the end, I managed to resolve it by using the automatic backup script. I used this fork by WaleedMortaja (https://gist.github.com/WaleedMortaja/ddc87fc4dc7dcb73178e971c299d7e78), which seems to work much better than the one mentioned in the guide and also works on MacOS with a small modification.

With this script, the backup size is only 49KB (I've been using Firefly for 2 months, so I have few transactions), while the one done with the manual backup is 7MB. Why is there such a difference? Could it be that the manual backup command is saving non-essential data? Can you take a look to see if it also gives you trouble when doing the backup and importing it into different systems (for example, from Linux to Windows or Mac)?

When restoring with the script, everything worked without any issues. 

Even though I think it would be useful to include an easy backup option directly within Firefly, maybe integrating the created script directly into the Firefly GUI. Think about it.

**Reply by JC5** [reply]
I will see if that script can be part of the documentation. But there will not be an integrated backup solution.

**Reply by Pindol83** [reply]
Ok, and could you explain why there's this difference in backup sizes between manually made backups and those created using the script?

**Reply by JC5** [reply]
I'm not sure. I guess the one backups the entire volume and the other only the sql export.
